pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -305,22 +481,94 @@ func configMapDeleted(obj interface{}) {
+	// Check if it is either internal or supervisor FSS configmap
+	if fssConfigMap.Name == k8sOrchestratorInstance.supervisorFSS.configMapName &&
+		fssConfigMap.Namespace == k8sOrchestratorInstance.supervisorFSS.configMapNamespace {
+		if getSvFssCRAvailability() {
+			log.Debugf("Ignoring supervisor FSS configmap delete event as %q CR is present", featurestates.CRDSingular)
+			return
 		}
-		log.Infof("configMapDeleted: %v deleted. Setting supervisor feature state values to false %v", fssConfigMap.Name,
-			k8sOrchestratorInstance.supervisorFSS.featureStates)
+		log.Errorf("configMap %q in namespace %q deleted. This is a system resource, kindly restore it.", fssConfigMap.Name, fssConfigMap.Namespace)
+		os.Exit(1)
+	} else if fssConfigMap.Name == k8sOrchestratorInstance.internalFSS.configMapName &&
+		fssConfigMap.Namespace == k8sOrchestratorInstance.internalFSS.configMapNamespace {
+		log.Errorf("configMap %q in namespace %q deleted. This is a system resource, kindly restore it.", fssConfigMap.Name, fssConfigMap.Namespace)
+		os.Exit(1)
 	}
-	// Internal FSS configmap
-	if fssConfigMap.Name == k8sOrchestratorInstance.internalFSS.configMapName && fssConfigMap.Namespace == k8sOrchestratorInstance.internalFSS.configMapNamespace {
-		for featureName := range k8sOrchestratorInstance.internalFSS.featureStates {
-			k8sOrchestratorInstance.internalFSS.featureStates[featureName] = strconv.FormatBool(false)
-		}
-		log.Infof("configMapDeleted: %v deleted. Setting internal feature state values to false %v", fssConfigMap.Name,
-			k8sOrchestratorInstance.internalFSS.featureStates)
+}
+
+// fssCRAdded adds supervisor feature state switch values from the cnscsisvfeaturestate CR
+func fssCRAdded(obj interface{}) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = logger.NewContextWithLogger(ctx)
+	log := logger.GetLogger(ctx)
+
+	var svFSSObject featurestatesv1alpha1.CnsCsiSvFeatureStates
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.(*unstructured.Unstructured).Object, &svFSSObject)
+	if err != nil {
+		log.Errorf("fssCRAdded: failed to cast object to %s. err: %v", featurestates.CRDSingular, err)
+		return
+	}
+	if svFSSObject.Name != featurestates.SVFeatureStateCRName {
+		log.Warnf("fssCRAdded: Ignoring %s CR object with name %q", featurestates.CRDSingular, svFSSObject.Name)
+		return
 	}
+	setSvFssCRAvailability(true)
+	for _, fss := range svFSSObject.Spec.FeatureStates {
+		k8sOrchestratorInstance.supervisorFSS.featureStates[fss.Name] = strconv.FormatBool(fss.Enabled)
+	}
+	log.Infof("fssCRAdded: New supervisor feature states values stored successfully from %s CR object: %v",
+		featurestates.SVFeatureStateCRName, k8sOrchestratorInstance.supervisorFSS.featureStates)
+}
+
+// fssCRUpdated updates supervisor feature state switch values from the cnscsisvfeaturestate CR
+func fssCRUpdated(oldObj, newObj interface{}) {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Can this line to line 544 but refactored to a function since its being used in all three event handlers for fss CRs?

Reply:
It is only being repeated in Create and Update handlers but I cannot unite them as Divyen suggested we have logs which mention exactly what event we are handling. Each of the logs in these handlers have the handler name in the beginning.